Sources/GRPC/ClientConnection.swift

```diff
@@ -247,6 +257,40 @@ public enum ConnectionTarget {
   }
 }
 
+/// The policy used to determine how an RPC should be started in relation to the connection it
+/// will use.
+public struct CallStartPolicy: Hashable {
+  internal enum Policy: Hashable {
+    case waitForReadyConnection
+    case fastFailure
+  }
```

Comment:
Very much open to naming suggestions here.

Reply:
The prior art from Apple platforms is `waitsForConnectivity`, so we could investigate something in this vein.

Reply:
I was concerned about overloading waiting for connectivity: in this context we might try to establish a connection a number of times during which – if adopted – we could see many `WaitingForConnectivity` events. If we later surface an option to disable waiting via NIOTS then we _could_ end up with a third policy which is somewhere between the two. 

However, thinking some more, disabling waiting for connectivity via NIOTS should just be done when applicable if the fast failure policy is being used. Observing the event in the connectivity delegate serves as extra information which _could_ have been used to cancel an RPC: however there would be no need to action on it to cancel RPCs since `fastFailure` would give you that behaviour anyway.

So, naming:

- `waitsForConnectivity`
- `fastFailure`

I'm also not super keen on `CallStartPolicy`, `Behaviour` seems like a better fit than `Policy`. WDYT?

Reply:
Yeah, I like `Behaviour` better. The rest of your proposal sounds great